src/qililab/digital/circuit_transpiler.py

```diff
@@ -37,20 +43,26 @@ class CircuitTranspiler:
     - ``transpile_circuit``: runs both of the methods above sequentially
 
     Args:
-        platform (Platform): platform object containing the runcard and the chip's physical qubits.
+        settings (DigitalCompilationSettings | Platform): Object containing the digital compilations settings and the info on chip's physical qubits.
+            It can be both directly the `DigitalCompilationSettings` object or a `Platform` object with the settings defined in it.
     """
 
-    def __init__(self, digital_compilation_settings: DigitalCompilationSettings):  # type: ignore # ignore typing to avoid importing platform and causing circular imports
-        self.digital_compilation_settings = digital_compilation_settings
+    def __init__(self, settings: Union[DigitalCompilationSettings, "Platform"]):
+        if isinstance(settings, DigitalCompilationSettings):
+            self.settings: DigitalCompilationSettings = settings
+        elif hasattr(settings, "digital_compilation_settings") and settings.digital_compilation_settings is not None:
+            self.settings = settings.digital_compilation_settings
+        else:
+            raise ValueError("`setting`s must be a `DigitalCompilationSettings` or a `Platform` with them defined.")
 
     def transpile_circuits(
         self,
         circuits: list[Circuit],
-        placer: Placer | type[Placer] | tuple[type[Placer], dict] | None = None,
-        router: Router | type[Router] | tuple[type[Router], dict] | None = None,
+        placer: Union["Placer", type["Placer"], tuple[type["Placer"], dict], None] = None,
+        router: Union["Router", type["Router"], tuple[type["Router"], dict], None] = None,
```

Comment:
In general we prefer the `|` notation. If you get conflicts because of the strings add `from __future__ import annotations` as the first import and you don't need the strings any more 😄

Reply:
Although I think you shouldn't even need the annotations import in this case. Since it is external class (usually you need it for self-referencing).

Reply:
Ohh nice to know! I'll change it right away 👌 

I thought the `from __future__ import annotations` only worked for classes defined in the same file, great!